docs/tutorials/data_management.md

```diff
@@ -0,0 +1,243 @@
+
+This tutorial introduces data providers beyond `icu_testdata`, as well as the `icu_datagen` tool.
+
+# 1. Prerequesites
+
+This tutorial assumes you have finished the [introductory tutorial](intro.md) and continues where that tutorial left off. In particular, you should still have the latest version of code for `myapp`.
+
+# 2. Generating data
+
+Data generation is done using the `icu_datagen` crate, which pulls in data from [Unicode's *Common Locale Data Repository* (*CLDR*)](http://cldr.unicode.org/index/downloads) and from `ICU4C` releases to generate `ICU4X` data. The crate has a command line interface as well as a Rust API, which can be used for example in `build.rs` files. Here we will use the CLI.
+
+First we will need to install the binary:
+
+```console
+$ cargo install icu_datagen --features bin
+```
+
+Get a coffee, this might take a while ☕.
+
+Once installed, run:
+
+```console
+$ icu4x-datagen --cldr-tag latest --icuexport-tag latest --out my-data --format blob --all-keys --all-locales
+```
+
+Let's dissect this invocation:
+* `--cldr-tag` selects the CLDR version to use
+* `--icuexport-tag` selects the ICU-exported data version to use
+* `--out` is the location where we want the generated ICU4X data to be stored
+* `--format` sets the format of the output (we'll discuss formats later)
+* `--all-keys` `--all-locales` specifies that we want to include all data for all locales
+
+This will generate a `my-data` file containing the serialized data.
+
+# 3. Using the generated data
+
+Once we have generated some data, it needs to be loaded as a data provider. The blob format we chose can be loaded by `BlobDataProvider` from the `icu_provider_blob` crate.
+
+This provider performs deserialization, so it's a `BufferProvider`. This means that the feature `"serde"` needs to be enabled on `icu`.
+
+Let's update our `Cargo.toml`:
+
+```console
+$ cargo add icu --features serde
+$ cargo add icu_provider_blob
+```
+
+We can then use the provider in our code:
+
+```rust
+use icu::locid::{locale, Locale};
+use icu::calendar::DateTime;
+use icu::datetime::{DateTimeFormatter, options::length};
+use icu_provider_blob::BlobDataProvider;
+
+const LOCALE: Locale = locale!("ja");
+
+fn main() {
+    let blob = std::fs::read("my-data").expect("Failed to read file");
+    let buffer_provider = 
+        BlobDataProvider::try_new_from_blob(blob.into_boxed_slice())
+            .expect("Failed to initialize Data Provider.");
+
+    let options = length::Bag::from_date_time_style(length::Date::Long, length::Time::Medium);
+
+    let dtf = DateTimeFormatter::try_new_with_buffer_provider(&buffer_provider, &LOCALE.into(), options.into())
+        .expect("Failed to initialize DateTimeFormatter");
+
+    let date = DateTime::try_new_gregorian_datetime(2020, 10, 14, 13, 21, 28)
```

Comment:
Issue: As in the other tutorial, don't mix `try_new_gregorian_datetime` and `DateTimeFormatter`